### DISCUSSION TITLE
Compatibility with pre-existing systems

### DISCUSSION THREAD

**Discussion by catharsis71** [swizzin/swizzin#654] [Q&A + Help] [answered]
I had a bad experience with a script similar to this one so I figured I should ask some questions first.

Is this safe to install on a live system already being used as a web server / etc?

Is there any risk of disruption to existing services (specifically Apache)?

Is it easy to uninstall and revert ALL changes if something goes wrong or if I just don't like it?

**Comment by flying-sausages** [comment, marked as answer]
Apache is the one thing that swizzin is **explicitly** incompatible with. In all honesty, depending on your use case, nginx might be a better solution compared to Apache 80% of the time anyway, so I would suggest considering a migration.

As far as uninstalling goes, swizzin itself makes little changes to the system. It will make some minor permission changes to the master user if it already existed before, so i would suggest making a separate user. 

All applications have uninstall scripts available, but some of them will preserve the application configuration so that you can come back to it later if you want to. It would never leave processes running behind that would cause some issues for you.

We always suggest users use a clean, fresh machine for the best results. With that said, if you weren't using Apache, you would probably not notice any difference.

Feel free to come to the Discord and discuss your specific use-case and what your best solution would be!

P.S., I took the liberty of changing the title to something a little less... alarming :)
